src/makeCalculator.js

```diff
@@ -37,7 +37,37 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  return {
+    result: 0,
+
+    add(a, b) {
+      return a + b;
+    },
+
+    subtract(a, b) {
+      return a - b;
+    },
+
+    multiply(a, b) {
+      return a * b;
+    },
+
+    divide(a, b) {
+      return a / b;
+    },
+
+    operate(callback, num) {
+      this.result = callback(this.result, num);
+
+      return this;
+    },
```

Comment:
```suggestion
    operate(callback, num) {
      operate(callback, num) {
      callback.call(this, num);

      return this;
    },
```
What about your question on Q&A, you can use `call` to set the context manually. Then you would be able to use `this` in methods ` add, subtract ...`